src/Sarif.Viewer.VisualStudio/Models/FixModel.cs

```diff
@@ -62,240 +45,9 @@ public ObservableCollection<ArtifactChangeModel> ArtifactChanges
-
-                    IEnumerable<ArtifactChangeModel> artifactChanges = selectedFix.ArtifactChanges.Where(fc => fc.FilePath.Equals(file, StringComparison.OrdinalIgnoreCase));
-                    foreach (ArtifactChangeModel fileChange in artifactChanges)
-                    {
-                        replacements.AddRange(fileChange.Replacements);
-                    }
-
-                    byte[] fixedFile;
-                    if (TryFixFile(file, replacements, true, out fixedFile))
-                    {
-                        string extension = Path.GetExtension(file);
-                        string baseFileName = Path.GetFileNameWithoutExtension(file);
-                        string previewFileName = baseFileName + ".preview" + extension;
-                        string previewFilePath = Path.Combine(Path.GetTempPath(), previewFileName);
-
-                        File.WriteAllBytes(previewFilePath, fixedFile.ToArray());
-
-                        var dte = AsyncPackage.GetGlobalService(typeof(DTE)) as DTE2;
-
-                        dte.ExecuteCommand("Tools.DiffFiles",
-                                            $"\"{file}\" " +
-                                            $"\"{previewFilePath}\" " +
-                                            $"\"{Resources.FixPreviewWindow_OriginalFileTitle}\" " +
-                                            $"\"{Resources.FixPreviewWindow_PreviewFixedFileTitle}\"");
-}
-                }
-            }
-        }
-
-        internal void Apply()
-        {
-            HashSet<string> files = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-
-            // Gather the list of files for the fix.
-            foreach (var fileChanges in ArtifactChanges)
-            {
-                files.Add(fileChanges.FilePath.ToLower());
-            }
-
-            foreach (string file in files)
-            {
-                if (_fileSystem.FileExists(file))
-                {
-                    List<ReplacementModel> replacements = new List<ReplacementModel>();
-
-                    var fileChanges = ArtifactChanges.Where(fc => fc.FilePath.Equals(file, StringComparison.OrdinalIgnoreCase));
-                    foreach (ArtifactChangeModel fileChange in fileChanges)
-                    {
-                        replacements.AddRange(fileChange.Replacements);
-                    }
-
-                    byte[] fixedFile;
-                    if (TryFixFile(file, replacements, false, out fixedFile))
-                    {
-                        _fileSystem.WriteAllBytes(file, fixedFile.ToArray());
-                        s_sourceFileFixLedger[file].LastModified = File.GetLastWriteTime(file);
-
-                        // Save after every fix because we don't have sufficient events to
-                        // guarantee data integrity
-                        SaveFixLedger();
-                    }
-                }
-            }
-
             FixApplied?.Invoke();
```

Comment:
>FixApplied [](start = 12, length = 10)

`SarifErrorListItem` subscribes to this event. When it fires, the list item can mark itself as "fixed" so it doesn't offer the light bulb again for that error.